wicker/core/column_files.py

```diff
@@ -196,37 +251,38 @@ def __init__(
         :param filelock_timeout_seconds: number of seconds after which to timeout on waiting for downloads,
             defaults to -1 which indicates to wait forever
         :type filelock_timeout_seconds: int, optional
+        :param path_factory: Path factory for determining paths to col files from root
+        :type path_factory: WickerPathFactory, defaults to S3PathFactory
+        :param storage: Storage used for grabbing files. Defaults to nont and creates S3DataStorage if none.
+        :type storage: Optional[AbstractDataStorage]
         :param dataset_name: name of the dataset, defaults to None
         :type dataset_name: str, optional
         """
-        self._s3_storage = storage if storage is not None else S3DataStorage()
+        super().__init__(dataset_name=dataset_name, path_factory=path_factory)
+        self._storage = storage if storage is not None else S3DataStorage()
         self._root_path = local_cache_path_prefix
         self._filelock_timeout_seconds = filelock_timeout_seconds
-        self._columns_root_path = path_factory.get_column_concatenated_bytes_files_path(dataset_name=dataset_name)
 
-    def read(
-        self,
-        cbf_info: ColumnBytesFileLocationV1,
-    ) -> bytes:
-        column_concatenated_bytes_file_path = os.path.join(self._columns_root_path, str(cbf_info.file_id))
+    def read(self, column_bytes_file_info: ColumnBytesFileLocationV1) -> bytes:
+        """Read data from column bytes file.
+
+        :param column_bytes_file_info: Location object to column bytes file on data store
+        :type column_bytes_file_info: ColumnBytesFileLocationV1
+        :return: bytes read from the column bytes file
+        """
+        column_concatenated_bytes_file_path = os.path.join(
+            self._column_bytes_root_path, str(column_bytes_file_info.file_id)
+        )
 
-        local_path = self._s3_storage.fetch_file(
+        local_path = self._storage.fetch_file(
             column_concatenated_bytes_file_path,
             self._root_path,
             timeout_seconds=self._filelock_timeout_seconds,
         )
 
-        with open(local_path, "rb") as f:
-            f.seek(cbf_info.byte_offset)
-            return f.read(cbf_info.data_size)
-
```

Comment:
We moved the `resolve_pointers` function below to the `ColumnBytesFileReader` base class with no changes (since that class needs this function as well).